stubs/toml/toml/decoder.pyi

```diff
@@ -0,0 +1,79 @@
+import sys
+from _typeshed import SupportsRead
+from typing import Any, Callable, Generic, MutableMapping, Pattern, Protocol, Text, TypeVar, overload
+
+_MutableMappingT = TypeVar("_MutableMappingT", bound=MutableMapping[str, Any])
+_KT_contra = TypeVar("_KT_contra", contravariant=True)
+_VT = TypeVar("_VT")
+
+if sys.version_info >= (3, 0):
+    from pathlib import PurePath
+
+    FNFError = FileNotFoundError
+    _PathLike = str | bytes | PurePath
+else:
+    FNFError = IOError
+    _PathLike = str | bytes
+
+class _SupportsGetSetItem(Protocol[_KT_contra, _VT]):
+    def __getitem__(self, __k: _KT_contra) -> _VT: ...
+    def __setitem__(self, __k: _KT_contra, __v: _VT) -> None: ...
+
+TIME_RE: Pattern[str]
+
+class TomlDecodeError(ValueError):
+    msg: str
+    doc: str
+    pos: int
+    lineno: int
+    colno: int
+    def __init__(self, msg: str, doc: str, pos: int) -> None: ...
+
+class CommentValue(_SupportsGetSetItem[_KT_contra, _VT]):
+    val: _SupportsGetSetItem[_KT_contra, _VT]
+    comment: str
+    def __init__(
+        self, val: _SupportsGetSetItem[_KT_contra, _VT], comment: str, beginline: bool, _dict: type[_MutableMappingT]
```

Comment:
- Same `val` change here
- Using a `TypeVar` only once in a function definition doesn't make sense. They are meant to be used when two different things must have the same type.

```suggestion
        self, val: Any, comment: str, beginline: bool, _dict: type[MutableMapping[str, Any]]
```